### PR TITLE
Add org.b3log.siyuan

### PR DESCRIPTION
**Pull request by proletarius101**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/siyuan-note/siyuan/issues/6877#issuecomment-1812811001
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flatpak.org/en/latest/first-build.html

### CODE REVIEW

File: org.b3log.siyuan.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>org.b3log.siyuan</id>
+
+  <name>SiYuan</name>
+  <summary>A privacy-first personal knowledge management system, support fine-grained block-level
+    reference and Markdown WYSIWYG.</summary>
```
Comment (hfiguiere) [inline org.b3log.siyuan.metainfo.xml:7]:
```suggestion
    reference and Markdown WYSIWYG</summary>
```

File: org.b3log.siyuan.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        done
+
+      - install -Dm644 "${FLATPAK_DEST}/siyuan/resources/stage/icon-large.png" "${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png"
+      - install -Dm644 org.b3log.siyuan.desktop "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+      - install -Dm644 org.b3log.siyuan.metainfo.xml /app/share/metainfo/$FLATPAK_ID.metainfo.xml
+    sources:
+      - type: archive
+        dest: siyuan
+        only-arches: [x86_64]
+        url: https://github.com/siyuan-note/siyuan/releases/download/v2.10.14/siyuan-2.10.14-linux.tar.gz
```
Comment (hfiguiere) [inline org.b3log.siyuan.yml:39]: so how does this work on the aarch64 flatpak that is being built?

can't it just be built from source instead?

File: org.b3log.siyuan.desktop [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
```
Comment (bbhtt) [inline org.b3log.siyuan.desktop:1]: Can you submit this upstream?

File: org.b3log.siyuan.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,84 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bbhtt) [inline org.b3log.siyuan.metainfo.xml:1]: This too

File: org.b3log.siyuan.yml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    sources:
+      - type: script
+        dest-filename: start-siyuan.sh
+        commands:
+          - set -oue pipefail
+
+          - export FLATPAK_ID="${FLATPAK_ID:-org.b3log.siyuan}"
+          - export TMPDIR="${XDG_RUNTIME_DIR}/app/${FLATPAK_ID}"
+
+          - zypak-wrapper /app/siyuan/siyuan --ozone-platform-hint=wayland --enable-features=UseOzonePlatform,WaylandWindowDecorations,Vulkan
```
Comment (bbhtt) [inline org.b3log.siyuan.yml:66]: I'd suggest not enabling these by default, you can pass auto to the first argument which will choose wayland if the desktop session is wayland, then fallback to xwayland/x window.

**Comment by hfiguiere** [comment]
bot, build org.b3log.siyuan

**Comment by hfiguiere** [comment]
```
org.b3log.siyuan:                  AppData problem: tag-invalid : <screenshot> content was duplicated
org.b3log.siyuan:                  AppData problem: tag-invalid : <url> does not start with 'http://' [>https://github.com/siyuan-note/siyuan/issues]
```

File: flathub.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+    "automerge-flathubbot-prs": true,
```
Comment (bbhtt) [inline flathub.json:2]: This is a bad idea, builds will get merged without any testing.

File: org.b3log.siyuan.desktop (hunk 3 above)
Comment (bbhtt) [reply org.b3log.siyuan.desktop:1]: So the appimages come without a desktop file?

File: org.b3log.siyuan.yml (hunk 5 above)
Comment (bbhtt) [reply org.b3log.siyuan.yml:66]: You need `--socket=x11` probably

Comment (hfiguiere) [reply org.b3log.siyuan.yml]: in general wayland support in Electron is poor and therefor should be avoided.

**Comment by bbhtt** [comment]
I suggest dropping the gtk4 flag too. Electron doesn't support it and still depends on Gtk3.

https://github.com/electron/electron/issues/33690

Loading both in the same process will lead to crashes.

**Pull request closed without merging**